portal-api/src/v1/validation/utilisation-report/utilisation-report-upload-validator.ts

```diff
@@ -0,0 +1,21 @@
+import { ReportPeriod, UTILISATION_REPORT_RECONCILIATION_STATUS, getFormattedReportPeriodWithLongMonth } from '@ukef/dtfs2-common';
+import { InvalidReportStatusError } from '../../errors';
+import api from '../../api';
+
+export const validateReportForPeriodIsInReportNotReceivedStateAndReturnId = async (bankId: string, reportPeriod: ReportPeriod): Promise<number> => {
```

Comment:
This feels to me like it is doing too much for one function 🤔

Reply:
Yeah i struggled with this one a bit, I think I should perhaps move most of it back to the controller